scripts/reactor_deployment.py

```diff
@@ -0,0 +1,496 @@
+        number is set or varies based on time.
+    """
+    # Initialize the number of reactor columns.
+    for reactor in ar_dict.keys():
+        if f'num_{reactor}' not in df:
+            df[f'num_{reactor}'] = 0
+        else:
+            pass
+
+    # First we will apply the rough random.
+    df = rand_deployment(df, base_col, ar_dict, set_seed, rough=True)
+
+    # Now we will make a remaining cap column.
+    df['remaining_cap'] = df[base_col] - df['new_cap']
+
+    # make columns for the randomly deployed reactors and the greedy reactors
+    for reactor in ar_dict.keys():
+        df[f'greedy_num_{reactor}'] = 0
+        df[f'rand_num_{reactor}'] = df[f'num_{reactor}']
+
+    # Now we will use the remaining cap column with a greedy deployment.
+    df = greedy_deployment(df, 'remaining_cap', ar_dict)
+
+    # reset the total capacity column
+    df['new_cap'] = 0
+
+    # populate the greedy reactor column
+    for year in range(len(df[base_col])):
+        for reactor in ar_dict.keys():
+            df.loc[year, f'greedy_num_{reactor}'] = \
+                   df.loc[year, f'num_{reactor}'] \
+                   - df.loc[year, f'rand_num_{reactor}']
+            df.loc[year, f'new_cap'] += df.loc[year, f'{reactor}_cap']
+
+    return df
+
+
+# # # # # # # # # # # # Analysis Functions # # # # # # # # # # # #
+# analyze how well each method did with the amount and percent of over/
+# under-prediction.
+
+def simple_diff(df, base, proj):
+    """
+    Calculate the difference between the projected and base capacities.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        The output pandas DataFrame from the deployment functions.
+    base: str
+        The name of the base capacity column in the DataFrame.
+    proj: str
+        The name of the projected capacity column in the DataFrame.
+    """
+    return df[proj] - df[base]
+
+
+def calc_percentage(df, base):
+    """
+    Calculate the percentage difference between proj and base.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        The output pandas DataFrame from the deployment functions.
+    base: str
+        The name of the base capacity column in the DataFrame.
+    """
+    return (df['difference'] / df[base]) * 100
+
+
+def analyze_algorithm(df, base, proj, ar_dict):
+    """
+    This function takes in a DataFrame output of the deployment functions
+    above, and returns a series of metrics so you can compare different
+    deployments.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        The output pandas DataFrame from the deployment functions.
+    base: str
+        The name of the base capacity column in the DataFrame.
+    proj: str
+        The name of the projected capacity column in the DataFrame.
+    ar_dict: dictionary
+        A dictionary of reactors with information of the form:
+        {reactor: [Power (MWe), capacity_factor (%), lifetime (yr),
+        [distribution]]}.
+
+    Returns
+    -------
```

Comment:
It returns a dictionary containing these

Reply:
Oh yes, that's a good distinction!